File: app/src/main/java/ru/michanic/mymot/UI/Adapters/ImagesSliderAdapter.java
package ru.michanic.mymot.UI.Adapters;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.List;

import ru.michanic.mymot.UI.Cells.ImagesSliderSlide;

public class ImagesSliderAdapter extends FragmentPagerAdapter {

    List<String> images;

    public ImagesSliderAdapter(FragmentManager fm, List<String> images) {
        super(fm);
        this.images = images;
    }

    @Override
    public Fragment getItem(int i) {
        return ImagesSliderSlide.newInstance(images.get(i));
    }

    @Override
    public int getCount() {
        return images.size();
    }
}

File: app/src/main/java/ru/michanic/mymot/Kotlin/UI/Adapters/ImagesSliderAdapter.kt
package ru.michanic.mymot.Kotlin.UI.Adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import ru.michanic.mymot.Kotlin.UI.Cells.ImagesSliderSlide

class ImagesSliderAdapter(fm: FragmentManager, var images: List<String>) :
    FragmentPagerAdapter(fm) {
    override fun getItem(i: Int): Fragment {
        return ImagesSliderSlide.newInstance(images[i])
    }

    override fun getCount(): Int {
        return images.size
    }
}